File: ts/test/entity/search_addresses_get/SearchAddressesGetEntity.test.ts

const envlocal = __dirname + '/../../../.env.local'
require('dotenv').config({ quiet: true, path: [envlocal] })

import Path from 'node:path'
import * as Fs from 'node:fs'

import { test, describe, afterEach } from 'node:test'
import assert from 'node:assert'


import { AddressLookupServiceSDK, BaseFeature, stdutil } from '../../..'

import {
  envOverride,
  liveDelay,
  makeCtrl,
  makeMatch,
  makeReqdata,
  makeStepData,
  makeValid,
  maybeSkipControl,
} from '../../utility'


describe('SearchAddressesGetEntity', async () => {

  // Per-test live pacing. Delay is read from sdk-test-control.json's
  // `test.live.delayMs`; only sleeps when ADDRESSLOOKUPSERVICE_TEST_LIVE=TRUE.
  afterEach(liveDelay('ADDRESSLOOKUPSERVICE_TEST_LIVE'))

  test('instance', async () => {
    const testsdk = AddressLookupServiceSDK.test()
    const ent = testsdk.SearchAddressesGet()
    assert(null != ent)
  })


  test('basic', async (t) => {

    const live = 'TRUE' === process.env.ADDRESS_LOOKUP_SERVICE_TEST_LIVE
    for (const op of ['list']) {
      if (maybeSkipControl(t, 'entityOp', 'search_addresses_get.' + op, live)) return
    }

    const setup = basicSetup()
    // The basic flow consumes synthetic IDs and field values from the
    // fixture (entity TestData.json). Those don't exist on the live API.
    // Skip live runs unless the user provided a real ENTID env override.
    if (setup.syntheticOnly) {
      t.skip('live entity test uses synthetic IDs from fixture — set ADDRESS_LOOKUP_SERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID JSON to run live')
      return
    }
    const client = setup.client
    const struct = setup.struct

    const isempty = struct.isempty
    const select = struct.select

    let search_addresses_get_ref01_data = Object.values(setup.data.existing.search_addresses_get)[0] as any

    // LIST
    const search_addresses_get_ref01_ent = client.SearchAddressesGet()
    const search_addresses_get_ref01_match: any = {}

    const search_addresses_get_ref01_list = await search_addresses_get_ref01_ent.list(search_addresses_get_ref01_match)


  })
})



function basicSetup(extra?: any) {
  // TODO: fix test def options
  const options: any = {} // null

  // TODO: needs test utility to resolve path
  const entityDataFile =
    Path.resolve(__dirname, 
      '../../../../.sdk/test/entity/search_addresses_get/SearchAddressesGetTestData.json')

  // TODO: file ready util needed?
  const entityDataSource = Fs.readFileSync(entityDataFile).toString('utf8')

  // TODO: need a xlang JSON parse utility in voxgig/struct with better error msgs
  const entityData = JSON.parse(entityDataSource)

  options.entity = entityData.existing

  let client = AddressLookupServiceSDK.test(options, extra)
  const struct = client.utility().struct
  const merge = struct.merge
  const transform = struct.transform

  let idmap = transform(
    ['search_addresses_get01','search_addresses_get02','search_addresses_get03'],
    {
      '`$PACK`': ['', {
        '`$KEY`': '`$COPY`',
        '`$VAL`': ['`$FORMAT`', 'upper', '`$COPY`']
      }]
    })

  // Detect whether the user provided a real ENTID JSON via env var. The
  // basic flow consumes synthetic IDs from the fixture file; without an
  // override those synthetic IDs reach the live API and 4xx. Surface this
  // to the test so it can skip rather than fail.
  const idmapEnvVal = process.env['ADDRESS_LOOKUP_SERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID']
  const idmapOverridden = null != idmapEnvVal && idmapEnvVal.trim().startsWith('{')

  const env = envOverride({
    'ADDRESS_LOOKUP_SERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID': idmap,
    'ADDRESS_LOOKUP_SERVICE_TEST_LIVE': 'FALSE',
    'ADDRESS_LOOKUP_SERVICE_TEST_EXPLAIN': 'FALSE',
  })

  idmap = env['ADDRESS_LOOKUP_SERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID']

  const live = 'TRUE' === env.ADDRESS_LOOKUP_SERVICE_TEST_LIVE

  if (live) {
    client = new AddressLookupServiceSDK(merge([
      {
      },
      extra
    ]))
  }

  const setup = {
    idmap,
    env,
    options,
    client,
    struct,
    data: entityData,
    explain: 'TRUE' === env.ADDRESS_LOOKUP_SERVICE_TEST_EXPLAIN,
    live,
    syntheticOnly: live && !idmapOverridden,
    now: Date.now(),
  }

  return setup
}
  

File: rb/test/search_addresses_get_entity_test.rb
# SearchAddressesGet entity test

require "minitest/autorun"
require "json"
require_relative "../AddressLookupService_sdk"
require_relative "runner"

class SearchAddressesGetEntityTest < Minitest::Test
  def test_create_instance
    testsdk = AddressLookupServiceSDK.test(nil, nil)
    ent = testsdk.SearchAddressesGet(nil)
    assert !ent.nil?
  end

  def test_basic_flow
    setup = search_addresses_get_basic_setup(nil)
    # Per-op sdk-test-control.json skip.
    _live = setup[:live] || false
    ["list"].each do |_op|
      _should_skip, _reason = Runner.is_control_skipped("entityOp", "search_addresses_get." + _op, _live ? "live" : "unit")
      if _should_skip
        skip(_reason || "skipped via sdk-test-control.json")
        return
      end
    end
    # The basic flow consumes synthetic IDs from the fixture. In live mode
    # without an *_ENTID env override, those IDs hit the live API and 4xx.
    if setup[:synthetic_only]
      skip "live entity test uses synthetic IDs from fixture — set ADDRESSLOOKUPSERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID JSON to run live"
      return
    end
    client = setup[:client]

    # Bootstrap entity data from existing test data.
    search_addresses_get_ref01_data_raw = Vs.items(Helpers.to_map(
      Vs.getpath(setup[:data], "existing.search_addresses_get")))
    search_addresses_get_ref01_data = nil
    if search_addresses_get_ref01_data_raw.length > 0
      search_addresses_get_ref01_data = Helpers.to_map(search_addresses_get_ref01_data_raw[0][1])
    end

    # LIST
    search_addresses_get_ref01_ent = client.SearchAddressesGet(nil)
    search_addresses_get_ref01_match = {}

    search_addresses_get_ref01_list_result = search_addresses_get_ref01_ent.list(search_addresses_get_ref01_match, nil)
    assert search_addresses_get_ref01_list_result.is_a?(Array)

  end
end

def search_addresses_get_basic_setup(extra)
  Runner.load_env_local

  entity_data_file = File.join(__dir__, "..", "..", ".sdk", "test", "entity", "search_addresses_get", "SearchAddressesGetTestData.json")
  entity_data_source = File.read(entity_data_file)
  entity_data = JSON.parse(entity_data_source)

  options = {}
  options["entity"] = entity_data["existing"]

  client = AddressLookupServiceSDK.test(options, extra)

  # Generate idmap via transform.
  idmap = Vs.transform(
    ["search_addresses_get01", "search_addresses_get02", "search_addresses_get03"],
    {
      "`$PACK`" => ["", {
        "`$KEY`" => "`$COPY`",
        "`$VAL`" => ["`$FORMAT`", "upper", "`$COPY`"],
      }],
    }
  )

  # Detect ENTID env override before envOverride consumes it. When live
  # mode is on without a real override, the basic test runs against synthetic
  # IDs from the fixture and 4xx's. Surface this so the test can skip.
  entid_env_raw = ENV["ADDRESSLOOKUPSERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID"]
  idmap_overridden = !entid_env_raw.nil? && entid_env_raw.strip.start_with?("{")

  env = Runner.env_override({
    "ADDRESSLOOKUPSERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID" => idmap,
    "ADDRESSLOOKUPSERVICE_TEST_LIVE" => "FALSE",
    "ADDRESSLOOKUPSERVICE_TEST_EXPLAIN" => "FALSE",
  })

  idmap_resolved = Helpers.to_map(
    env["ADDRESSLOOKUPSERVICE_TEST_SEARCH_ADDRESSES_GET_ENTID"])
  if idmap_resolved.nil?
    idmap_resolved = Helpers.to_map(idmap)
  end

  if env["ADDRESSLOOKUPSERVICE_TEST_LIVE"] == "TRUE"
    merged_opts = Vs.merge([
      {
      },
      extra || {},
    ])
    client = AddressLookupServiceSDK.new(Helpers.to_map(merged_opts))
  end

  live = env["ADDRESSLOOKUPSERVICE_TEST_LIVE"] == "TRUE"
  {
    client: client,
    data: entity_data,
    idmap: idmap_resolved,
    env: env,
    explain: env["ADDRESSLOOKUPSERVICE_TEST_EXPLAIN"] == "TRUE",
    live: live,
    synthetic_only: live && !idmap_overridden,
    now: (Time.now.to_f * 1000).to_i,
  }
end
